runtime/gc_vlhgc/VLHGCAccessBarrier.cpp

```diff
@@ -449,8 +449,9 @@ MM_VLHGCAccessBarrier::copyStringCritical(J9VMThread *vmThread, GC_ArrayObjectMo
 		functions->setNativeOutOfMemoryError(vmThread, 0, 0);
 	} else {
 		if (isCompressed) {
+			jchar *dataArr = *data;
```

Comment:
This variable could be declared earlier to capture the value returned from `jniArrayAllocateMemoryFromThread()` and used to write to `*data` and replace references to `*data` later in this function.